Add subcommand to render Ansible output directly into RST files (#397)
ansible-community/antsibull-docs: docs/ansible-output.md

## Metadata and code blocks

To know which code blocks to update and what playbook and environment variables to use, you need to provide a `ansible-output-data` directive before the actual code block:

```rst
.. ansible-output-data::

    env:
      ANSIBLE_STDOUT_CALLBACK: community.general.tasks_only
      ANSIBLE_COLLECTIONS_TASKS_ONLY_NUMBER_OF_COLUMNS: "90"
    playbook: |-
      - hosts: localhost
        gather_facts: false
        tasks:
          - name: Sort list by version number
            debug:
              var: ansible_versions | community.general.version_sort
            vars:
              ansible_versions:
                - '2.8.0'
                - '2.11.0'
                - '2.7.0'
                - '2.10.0'
                - '2.9.0'

.. code-block:: ansible-output

    TASK [Sort list by version number] ********************************************************
    ok: [localhost] => {
        "ansible_versions | community.general.version_sort": [
            "2.7.0",
            "2.8.0",
            "2.9.0",
            "2.10.0",
            "2.11.0"
        ]
    }
```

> The `ansible-output-data` directive results in no output when the `sphinx_antsibull_ext` Sphinx extension is used, and contains YAML data.

Reviewer: I think this is saying two related things but connects them in a way that is maybe harder to understand. At least for me.

The `ansible-output-data` directive does not generate any visible content in rendered documentation. The directive also contains YAML configuration data that is meant for `antsibull-docs` and not for readers.

> The `env` dictionary allows to set environment variables that are set when calling `ansible-playbook`.

Reviewer: ```suggestion
The `env` dictionary allows you to set environment variables that are set when calling `ansible-playbook`.
```

> In this example, we set an explicit callback stdout plugin (using `ANSIBLE_STDOUT_CALLBACK`),

Reviewer: ```suggestion
In this example, we set an explicit callback stdout plugin (using `ANSIBLE_STDOUT_CALLBACK`)
```

and provide configuration for that plugin (`ANSIBLE_COLLECTIONS_TASKS_ONLY_NUMBER_OF_COLUMNS`).

The playbook is provided as a multi-line YAML string.
Antsibull-docs looks for the next code block with language `ansible-output`, and replaces its contents with the output of `ansible-playbook playbook.yml`,

> where `playbook.yml` is filled with the provided playbook.

Reviewer: It honestly took me a couple of times to read this and then scroll back up and look through things before it really clicked. I think giving an example of the result (what gets generated for the reader) would be helpful and much more immediate.

Author: I tried to address this in ada87a707daa8c7e31c1fc0e5c8e4161c05e5676, PTAL.

Reviewer: Looks great!

Note that the lanuage for the code block can be overridden by providing `language`.

> Also you can prepend lines to the output using `prepend_lines`:

Reviewer: How about creating a table that groups the different configuration options or even a bulleted list:

- `env`
- `playbook`
- `language`
- `prepend_lines`

Author: I used a list in ada87a707daa8c7e31c1fc0e5c8e4161c05e5676, since tables with non-trivial content in MarkDown suck. I also added YAML comments to the example further below which demonstrates all these options.

(This is the directory that contains `galaxy.yml` or `MANIFEST.json`.)

It will check all `.rst` files in `docs/docsite/rst/`, if that directory exists,
and load configuration from `docs/docsite/config.yml`.
(See [more information on that configuration file](../collection-docs/#configuring-the-docsite).)
The configuration allows you to specify entries for `env` for all code blocks:

```yaml
---
# Configuration for 'antsibull-docs ansible-output'
ansible_output:
  # Insert definitions into 'env' for every ansible-output-data directive
  global_env:
    ANSIBLE_STDOUT_CALLBACK: community.general.tasks_only
    ANSIBLE_COLLECTIONS_TASKS_ONLY_NUMBER_OF_COLUMNS: 80
```

This is useful to standardize the callback and its settings for most code blocks in a collection's extra docs.

## Usage in CI

If you want to run `antsibull-docs ansible-output` in CI, you might find the `--check` parameter useful.
If that parameter is specified, antsibull-docs will not update files, but instead fail if a file would be modified.
A diff of the changes that would be applied will be printed to standard output.

!!! warning

>     Please note that you have to make sure that `antsibull-docs ansible-output` runs in CI with minimum number of privileges,

Reviewer: ```suggestion
    Please note that you have to make sure that `antsibull-docs ansible-output` runs in CI with the minimum set of privileges,
```

    since it can run **arbitrary code**!

    Someone can add a playbook to documentation that recursively deletes all files you have access to.
    If you run `antsibull-docs ansible-output` (with or without `--check`) on such a RST file without sufficient isolation,

>     all your files will be gone.

Reviewer: ```suggestion
    all your files will be deleted.
```

Reviewer: Yikes!
